fix webapp search and add selenium test case

rahulsayini opened the pull request:
closed Fix search webapp ui pull request and creating new pull request

rahulsayini committed:
fix webapp search and add selenium test case

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+    public void testSearchApp() throws InterruptedException {
+        clickTab("Configuration");
+        clickTab("Web Apps");
+        Thread.sleep(300);
```
Please remove Thread.sleep

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+        clickTab("Web Apps");
+        Thread.sleep(300);
+        driver.findElement(By.xpath("//button[span[text()='Add']]")).click();
+        Thread.sleep(1000); // give a little time for the first element to get into focus as required by how
```
This is not needed here. This was used in tests that bring up form or dialog whose first element gets the focus.

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+        driver.switchTo().activeElement().sendKeys("/zApp1");
+        driver.findElement(By.xpath("//div[contains(text(), '" + GROUP_NAME_1 + "')]/input")).click();
+        driver.findElement(By.xpath("//button[span[text()='Ok']]")).click();
+        Thread.sleep(300);
```
Please remove Thread.sleep

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+/**
+ * Created by RS045609 on 5/5/2017.
```
Put your name here

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+        clickTab("Configuration");
+        clickTab("Web Apps");
+        Thread.sleep(300);
+        driver.findElement(By.xpath("//button[span[text()='Add']]")).click();
```
Use this instead because after web apps is clicked, there might be a chance that the loading icon is still running by the time you do the click to add

    clickWhenReady(By.xpath("//button[span[text()='Add']]"));

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/testsuite/configuration/app/AppSearchTest.java:
```diff
@@ -0,0 +1,71 @@
+        driver.findElement(By.xpath("//button[span[text()='Add']]")).click();
+        Thread.sleep(300); // give a little time for the first element to get into focus as required by how
+        // "focus on the first element" was implemented here
+        driver.switchTo().activeElement().sendKeys(APP_NAME2);
```
Since we have the abstract methods available already since you merged master you can simplify your code e.g.

    sendKeys(Keys.TAB)

You can also simplify your clicks by using

    click(...)

jeddcuison committed:
Refactor Selenium test case

Details

- remove Thread.sleep where not needed
- put test data creation in init method
- put test data removal in destroy method
- use parent methods to simplify element events such as click, clickWhenReady, sendKeys etc...
- refresh the browser to display all rows again after filtering
- use click instead of waitClick when waitClick is not required
